src/wp-admin/includes/ajax-actions.php

```diff
@@ -3041,6 +3041,7 @@ function wp_ajax_query_attachments() {
 
 	$posts       = array_map( 'wp_prepare_attachment_for_js', $attachments_query->posts );
 	$posts       = array_filter( $posts );
+	update_post_parent_caches( $posts );
```

Comment:
To reduce the number of database queries in each request, this needs to run above the line containing `array_map( 'wp_prepare_attachment_for_js'...`.

The function `wp_prepare_attachment_for_js()` is where each attachment's parent post is queried.

I suggest moving this immediately after the line `$attachments_query = new WP_Query( $query );` and running it with the parameter `$attachments_query->posts` instead of `$posts`.

Reply:
Ok @peterwilsoncc 
I am making the change you suggested